core/state/statedb.go

```diff
@@ -824,6 +873,14 @@ func (s *StateDB) IntermediateRoot(deleteEmptyObjects bool) common.Hash {
 			s.prefetcher = nil // Pre-byzantium, unset any used up prefetcher
 		}()
 	}
+
+	if s.witness != nil {
+		// if witness building is enabled, collect storage access lists from
+		// tries that will not be committed: accounts with non-mutated storage
+		// and self-destructed accounts.
+		s.collectNonCommittedTrieAccessLists()
```

Comment:
Ideally, we would only collect these after kicking off the concurrent update workers for the main/account tries, instead of blocking on all the read prefetchers to complete as we do here.  However, the `StateDB.prefetcher` is unset after the end of this function call.